server/api/grid_apis.py

```diff
@@ -0,0 +1,59 @@
+from flask_restful import Resource, current_app, request
+from server.services.grid_service import GridService
+from server.services.authentication_service import token_auth, tm
+from server.models.dtos.grid_dto import GridDTO
+
+
+class IntersectingTilesAPI(Resource):
+    @tm.pm_only()
+    @token_auth.login_required
+    def put(self):
+        """
+        Gets the tiles intersecting the aoi
+        ---
+        tags:
+            - grid
+        produces:
+            - application/json
+        parameters:
+            - in: header
+              name: Authorization
+              description: Base64 encoded session token
+              required: true
+              type: string
+              default: Token sessionTokenHere==
+            - in: body
+              name: body
+              required: true
+              description: JSON object containing aoi and tasks
+              schema:
+                  properties:
+                      areaOfInterest:
+                          schema:
+                              $ref: "#/definitions/GeoJsonMultiPolygon"
+                      grid:
+                          schema:
+                              properties:
+                                  type:
+                                      type: string
+                                      default: FeatureCollection
+                                  features:
+                                      type: array
+                                      items:
+                                          schema:
+                                              $ref: "#/definitions/GeoJsonFeature"
+        responses:
+            200:
+                description: Intersecting tasks found successfully
+            500:
+                description: Internal Server Error
+        """
+        try:
+            json = request.get_json()
+            grid_dto = GridDTO(json)
```

Comment:
I would split getting the Grid DTO and calling the service into 2 steps.  Then u can validate the DTO